File: playground/javascript/vuln_flow_sqli.js
// VULNERABLE: tainted req.body flows into db.query()
// cwacs should detect CWACS_FLOW_JS_SQLI on the query line

async function getUser(req, db) {
  const userId = req.body.id;                            // source: tainted
  const sql = `SELECT * FROM users WHERE id=${userId}`; // propagation
  await db.query(sql);                                   // sink → CWACS_FLOW_JS_SQLI
}

File: playground/python/vuln_flow_sqli.py
# VULNERABLE: tainted user input flows into SQL execute()
# cwacs should detect CWACS_FLOW_PY_SQLI on the execute line

from flask import request
import sqlite3

def get_user(db):
    user_id = request.args.get("id")          # source: tainted
    query = "SELECT * FROM users WHERE id=" + user_id  # propagation
    db.execute(query)                          # sink → CWACS_FLOW_PY_SQLI
